### DISCUSSION TITLE
Crazyflie direction control

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#862] [Q&A]
Hi, I followed the instruction to let drone auntonomus fly around the circle. I know in this instruction that is use flow deck as sensor. but how can flow deck will estimate the x-y dircetion without loco position system or light house system ?

**Comment by knmcguire** [comment]
The flow sensor is able to sense flow, which together with height and some more variables can be directly translated to velocity. This transformation is all handled in the [state estimator onboard](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/state_estimators/) of the crazyflie (the extended kalman filter)

The kalman filter uses measurement models which explains this equation, which is given [here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/state_estimators/#flowdeck-measurement-model).

I also recommend [listening to this guest lecture](https://www.bitcraze.io/about/events/epfl2020/) I gave a while ago. Here I explain the stabilization modules within the [crazyflie](https://www.youtube.com/watch?v=BXjSgCwb_bM&t=95s) and later I explain about the measurement model of the flowdeck. I do not go into detail of what an extended kalman filter is, but the internet will provide for that if you search for it.

**Reply by CalvinKoung** [reply]
Ok, I got it. But if I equipped the both flow deck and loco deck they were conflict to estimate height each other ?

**Reply by knmcguire** [reply]
Only when they are completely off, and their measurement noise is considered the same. But we inputed the estimated measurement noise of the loco positioning system to be less accurate than the range sensor on the flowdeck or z-ranger, so the kalman filter sees that as mostly relying on the range sensors input. None the less, the height wil still be influenced a bit by the loco positioning system.

Please, as I said before, first read in into robotics sensor fusion and  kalman filters and how they work.

**Reply by CalvinKoung** [reply]
OK, thanks
